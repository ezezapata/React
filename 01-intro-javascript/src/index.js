// Cuando hay una exportaciÃ³n por defecto no es necesario utilizar las llaves {} en el import
import heroes, {owners} from './data/heroes';


const getHeroeById = id => heroes.find(h => h.id === id);
const getHeroeByOwner = owner => heroes.filter(h => h.owner === owner);

console.log(getHeroeById(3));
console.log(getHeroeByOwner('Marvel'));