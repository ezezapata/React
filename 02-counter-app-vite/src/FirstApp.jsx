// Si la función es una promesa, React va a devolver un error porque las promesas son objetos
// const getObjInfo = ({message, title}) => {
//     return [message, title];
// }

// Si no estamos cambiando el valor de newMessage y no tiene relación con hooks, etc, entonces no es recomendado colocarlo dentro del functional component

// Tener en cuenta que en las Functional Component no podemos pasarle objetos a los elementos hijos, es decir, dentro de {} pero si podemos pasarle arreglos []
// const obj = {
//     message: 'Este es el cuerpo del mensaje',
//     title: 'Este es el encabezado',
// }
// export const FirstApp = () => {
//     return(// siempre tiene que estar el nodo padre
//         <> 
//             <h1>{ getObjInfo(obj) }</h1>
//             {/* {<code>JSON.stringify( obj )</code>} */}
//             <p>Que pasa!</p>
//         </>
//     )
// }

import PropTypes from 'prop-types';
// props es info que fluye desde el componente padre hacia este componente "FirstApp"
export const FirstApp = ({title, subTitle}) => {

    return (
        <>
            <h1 data-testid='test-title' >{title}</h1>
            <h2>{subTitle}</h2>
        </>
    )   
}

// Esta es la forma de agregarle tipado a react sin TypeScript
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

// Asignar valores por defecto a las props
/* FirstApp.defaultProps = {
    title: 'No hay título.',
    subTitle: 'No hay subtítulo.'
} */