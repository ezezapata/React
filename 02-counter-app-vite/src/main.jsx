// Importaciones que sirven para renderizar la app.
// import { HelloWorldApp } from './HelloWorldApp';
// import { FirstApp } from './FirstApp';
import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css'; // Esta es la forma de agregar estilos de manera global
import {CounterApp} from './01-CounterApp/CounterApp';
import {CounterApp2} from './01-CounterApp/CounterApp2';
import {CounterApp3} from './01-CounterApp/CounterApp3';
import { FirstApp } from './FirstApp';

// Functional components
// Los componentes basados en funciones se escriben con PascalCase
// Normalmente los componentes no estan en este archivo.
// Se encuentra en el archivo HelloWorldApp.jsx
// function App() {
//     return (<h1>Hola Mundo</h1>);
// };

ReactDOM.createRoot( document.getElementById('root') )
    .render(
        <React.StrictMode>
            {/* <FirstApp/> */}
            {/* <CounterApp value={0}/>
            <CounterApp2 value={3}/>
            <CounterApp3 number={0}/> */}
            <FirstApp />
        </React.StrictMode>
    ) 