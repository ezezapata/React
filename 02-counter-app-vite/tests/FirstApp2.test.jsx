import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas FirstApp2 Component', () => {
    const title = 'Titulo';
    const subTitle = 'SubTítulo';
    
    test('Debe hacer match con el snapshot', () => {
        const { container } = render( <FirstApp title={ title } subTitle={ subTitle } /> );
        expect( container ).toMatchSnapshot();
    });

    test( 'Debe mostrar el mensaje "SubTítulo"', () => {
        render( <FirstApp title= { title } /> );
        expect( screen.getByText( title ) ).toBeTruthy();
    });

    test( 'Debe mostrar el título en un h1', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByRole( 'heading', {level: 1} ).innerHTML ).toContain( title );
    });

    test( 'Debe mostrar el subtítulo enviado por las props', () => {
        render( <FirstApp title={ title } subTitle={ subTitle } /> );
        expect( screen.getAllByText( subTitle ).length ).toBe( 1 );
    });

});