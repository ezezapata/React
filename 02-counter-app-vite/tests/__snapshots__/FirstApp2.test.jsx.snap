// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas FirstApp2 Component Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Titulo
  </h1>
  <h2>
    SubTÃ­tulo
  </h2>
</div>
`;
